src/core_atmosphere/physics/physics_wrf/module_bl_ugwp_gwdo.F

```diff
@@ -302,13 +354,34 @@ subroutine gwdo_ugwp(u3d,v3d,t3d,qv3d,p3d,p3di,pi3d,z,pblh,kpbl2d,br1,xland, &
                     ,its=its,ite=ite,kte=kte,kme=kte+1        &
                     ,errmsg=errmsg,errflg=errflg)
 
+!
+! Option to call non-stationary gravity wave drag
+!
+    if (ngw_scheme) then
+       call ugwpv1_ngw_run(xlatd=xlatd_hv,raincv=raincv_hv,rainncv=rainncv_hv  &
+                          ,ddy_j1tau=ddy_j1tau_hv,ddy_j2tau=ddy_j2tau_hv       &
+                          ,jindx1_tau=jindx1_tau_hv,jindx2_tau=jindx2_tau_hv   &
+                          ,r_DoY=r_DoY,kdt=itimestep,dtp=dt                    &
+                          ,ugrs=u3d_hv,vgrs=v3d_hv                             &
+                          ,tgrs=t3d_hv,q1=qv3d_hv,prsl=p3d_hv,prslk=pi3d_hv    &
+                          ,prsi=p3di_hv,zl=z_hv,zi=zi_hv,ntau_d2t=ntau_d2t     &
+                          ,days_limb=days_limb,tau_limb=tau_limb               &
+                          ,rublten=rublten_hv,rvblten=rvblten_hv               &
+                          ,rthblten=rthblten_hv,ugwp_diags=ugwp_diags          &
+                          ,dudt_ngw=dudt_ngw_hv,dvdt_ngw=dvdt_ngw_hv           &
+                          ,dtdt_ngw=dtdt_ngw_hv,its=its,ite=ite,levs=kte)
+    endif
+
+
+
     !  Variables that are INTENT(OUT) or INTENT(INOUT):
 
     !  3d, layers:
     do k = kts,kte
        do i = its,ite
           rublten(i,k,j) = rublten_hv(i,k)
           rvblten(i,k,j) = rvblten_hv(i,k)
+          rthblten(i,k,j)= rthblten_hv(i,k)
```

Comment:
plus, it looks like you are overwriting the tendency arrays, not adding to them, which to me is a bug.

Reply:
I agree it looks like a bug, but fortunately it isn't because in UGWP/bl_ugwp.F we are adding to the tendencies, not overwriting:

! rotate tendencies from zonal/meridional back to model grid
!
   do k = kts,kte
      do i = its,ite
         rublten(i,k) = rublten(i,k)+dudt(i,k)*cosa(i) + dvdt(i,k)*sina(i)
         rvblten(i,k) = rvblten(i,k)-dudt(i,k)*sina(i) + dvdt(i,k)*cosa(i)